smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/protocol/HttpProtocolUnitTestResponseGenerator.kt

```diff
@@ -80,8 +82,15 @@ open class HttpProtocolUnitTestResponseGenerator protected constructor(builder:
                     val inputShape = model.expectShape(it)
                     val inputSymbol = symbolProvider.toSymbol(inputShape)
 
+                    val labelMembers = inputShape.members().filter { it.hasTrait<HttpLabelTrait>() }
+
                     // invoke the DSL builder for the input type
-                    writer.write("val input = ${inputSymbol.name}{}")
+                    writer.withBlock("val input = #T {", "}", inputSymbol) {
+                        labelMembers.forEach { memberShape ->
```

Comment:
Question: I'm assuming this is because some `httpResponseTest` fails without it?

It looks like we set default values for string only, [httpLabel](https://awslabs.github.io/smithy/1.0/spec/core/http-traits.html#httplabel-trait) can target:

> structure > member[trait|required] :test(> :test(string, number, boolean, timestamp)

Wonder if we should nix setting a default value for string and just handle the possible cases inline here?

Reply:
We already provided default values for [all number types](https://github.com/awslabs/smithy-kotlin/blob/main/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/core/KotlinSymbolProvider.kt#L67-L82) and [boolean](https://github.com/awslabs/smithy-kotlin/blob/main/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/core/KotlinSymbolProvider.kt#L104-L105). We didn't previously set defaults for strings or timestamps and so I added one for string to get the tests to pass. I'm not sure what a sensible default would be for timestamp so I left it blank.

We certainly could handle the param values inline in the test generator but I wonder what else we plan to use default values for if not something like this.

Reply:
After more discussion we decided that the global `Symbol.defaultValue` wasn't the right place for this, especially since there's not really a valid default for strings or timestamps. Given that the _need_ for a default is local to the test generator, I'll move the _specification_ of those defaults there as well. Reverting changes to **SymbolExt.kt** and **KotlinSymbolProvider.kt**...